crux/models/resource.py

```diff
@@ -487,11 +389,10 @@ def add_labels(self, labels_dict):
         )
 
         if response_result:
-            for label_key, label_value in labels_dict.items():
-                self._labels[label_key] = label_value
-            return True
-        else:
-            return False
+            # Sync the latest data from API to prevent inconsistency
+            self.refresh()
```

Comment:
`add_label()` will need `self.refresh()` too (I don't seem to see it in the diff).